pkg/model/components/clusterautoscaler.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/loader"
+)
+
+// ClusterAutoscalerOptionsBuilder adds options for cluster autoscaler to the model
+type ClusterAutoscalerOptionsBuilder struct {
+	*OptionsContext
+}
+
+var _ loader.OptionsBuilder = &ClusterAutoscalerOptionsBuilder{}
+
+func (b *ClusterAutoscalerOptionsBuilder) BuildOptions(o interface{}) error {
+	clusterSpec := o.(*kops.ClusterSpec)
+	cas := clusterSpec.ClusterAutoscaler
+	if cas == nil || !fi.BoolValue(cas.Enabled) {
+		return nil
+	}
+
+	if cas.Expander == nil {
+		cas.Expander = fi.String("least-waste")
```

Comment:
Is there a reason for deviating from the CA and setting `least-waste` as the default expander rather than `random`?

Reply:
Think this was the default in the example I sourced from. No other specific reason.

Reply:
We should probably take the default from cluster-autoscaler itself. That appears to be `random`.